File: Python3/3074. Apple Redistribution into Boxes.py
class Solution:
    def minimumBoxes(self, apple: List[int], capacity: List[int]) -> int:
        ans = 0
        boxes = [-c for c in capacity]
        heapify(boxes)
        apples = sum(apple)

        while boxes and apples > 0:
            apples += heappop(boxes)
            ans += 1

        return ans

File: Java/3074. Apple Redistribution into Boxes.java
class Solution {
    public int minimumBoxes(int[] apple, int[] capacity) {
        Arrays.sort(capacity);
        int apples = 0, ans = 0;
        for (int a: apple) {
            apples += a;
        }

        for (int i = capacity.length - 1; i >= 0; i--) {
            apples -= capacity[i];
            ans++;
            if (apples <= 0) {
                break;
            }
        }

        return ans;
    }
}